### ISSUE TITLE
Move time to another task

### ISSUE THREAD

**Issue by jesperkdab** [KDAB/Charm#212]
Quite often I forget to change task when starting a new job. So today I need to do this:
1) stop the current task
2) change its end time to when I stopped working on what it was tracking
3) start the new task
4) change its start time to the end time of the previous one

It would be useful with a dialog which would help me with that. I imagine something like this:
1) click a magic key (or go through the menus) to bring up the dialog
2) choose the new task, and how many minutes I've already been working on it.

Cheers

**Comment by frankosterfeld** [comment]
An alternative would be a dialog that conveniently allows editing two adjacent events conveniently, where updating the start/end time of an event also updates the end/start time of its "neighbour". UI ideas welcome.

**Comment by seanharmer** [comment]
How about a graphical representation showing a timeline for the day (or several stacked to show the whole week.

Each task would be a block on the timeline. Dragging the centre of the block would move the block. Dragging the left edge would move the start time earlier or later. Dragging the right edge would move the end time. If two blocks are touching then dragging the border between them adjust the end time of rt. left block and the start time of the right block.

Cheers,

Sean

**Comment by dfaure-kdab** [comment]
IMHO that's a lot of development work for something that won't be practical to use, because we use a minute granularity so dragging something with the mouse will most likely not give enough precision. Aiming at 16:35 by dragging a mouse where each pixel maps to a minute is extremely annoying compared to just typing 16:35 (yes I'm using an extreme case to make my point, I know the number of pixels per minute will depend on screen size and task duration etc.). A dialog that asks for the cutoff time seems to me much simpler to implement -and- much simpler to use.

**Comment by jesperkdab** [comment]
I agree with David, that Sean's way is way to much work for too little gain.
Besides, Frank and my solutions attacks two different usecases:
Mine: I'm recording and realizes I should have changed tasks
Franks: I'm done recording and realizes later I should have changed tasks at a different point for two tasks in the past.
I think my usecase is much more common.

Please see attached image for a suggestion for how the dialog could look like.
![dialog](https://cloud.githubusercontent.com/assets/4677854/9191364/affe4e7c-4000-11e5-9988-941d1d1cd5eb.png)

**Comment by frankosterfeld** [comment]
My thinking was that a general editor would make it easier to fix the currently running task (and the previous one) and might be the more general solution to both problems, editing past and the current task.

**Comment by dfaure-kdab** [comment]
Not sure if it's clear to everyone, but Jesper's use case is actually easy to handle in the current Charm GUI, using these steps:
- stop the recording
- edit the last event in the event window
- change the end time to when it should have stopped
- create new event
- notice that it will default to start+end equal to when the previous task stopped
- click on "now" to adjust the end date
- close, start recording again

I was never bothered by this, because the "start after the previous task ends" and "now" features make it easy enough to handle.

**Comment by jesperkdab** [comment]
David: Fair enough, a bit easier than what I outlines. Still this happens a lot to me, so making it even easier would be great.

**Comment by tmcguire** [comment]
On a related note, I'd like a "Set start time to end time of previous task" context menu entry, for situations in which I started the task too late, but still stopped the previous task earlier.

In general, some "quick edit" actions for common things like this, Frank's and Jesper's workflow would be quite useful.
